msvc/scripts/genproj.cs

```diff
@@ -887,6 +945,26 @@ public VsCsproj Generate (string library_output, Dictionary<string,MsbuildGenera
 
 		bool basic_or_build = (library.Contains ("-basic") || library.Contains ("-build"));
 
+		// If an EXE is built with nostdlib, it won't work unless run with mono.exe. This stops our build steps
+		//  from working in visual studio (because we already replace @MONO@ with '' on Windows.)
+
+		if (Target != Target.Library)
+			StdLib = true;
+
+		// We have our target framework set to 4.5 in many places because broken scripts check for files with 4.5
+		//  in the path, even though we compile code that uses 4.6 features. So we need to manually fix that here.
+
+		if (fx_version == "4.5")
+			fx_version = "4.6.2";
+
+		// The VS2017 signing system fails to sign using this key for some reason, so for now,
```

Comment:
Not signing is actually a good choice, as really signing requires either exposing privates keys or using special hardware/service to do the signing.